x-pack/qa/repository-old-versions-7x/src/javaRestTest/java/org/elasticsearch/oldrepos7x/OldRepositoryAccessIT.java

```diff
@@ -0,0 +1,525 @@
+        searchResponse = search(
+            index,
+            SearchSourceBuilder.searchSource()
+                .query(QueryBuilders.matchQuery("val", num))
+                .runtimeMappings(Map.of("val", Map.of("type", "long"))),
+            requestOptions
+        );
+        try {
+            logger.info(searchResponse);
+            assertEquals(1, searchResponse.getHits().getTotalHits().value());
+            assertEquals(id, searchResponse.getHits().getHits()[0].getId());
+            assertEquals(sourceForDoc(num), searchResponse.getHits().getHits()[0].getSourceAsString());
+        } finally {
+            searchResponse.decRef();
+        }
+
+        if (sourceOnlyRepository == false) {
+            // search using reverse sort on val
+            searchResponse = search(
+                index,
+                SearchSourceBuilder.searchSource()
+                    .query(QueryBuilders.matchAllQuery())
+                    .sort(SortBuilders.fieldSort("val").order(SortOrder.DESC)),
+                requestOptions
+            );
+            try {
+                logger.info(searchResponse);
+                // check sort order
+                assertEquals(
+                    expectedIds.stream().sorted(Comparator.comparingInt(this::getIdAsNumeric).reversed()).toList(),
+                    Arrays.stream(searchResponse.getHits().getHits()).map(SearchHit::getId).toList()
+                );
+            } finally {
+                searchResponse.decRef();
+            }
+
+            // look up postings
+            searchResponse = search(
+                index,
+                SearchSourceBuilder.searchSource().query(QueryBuilders.matchQuery("test", "test" + num)),
+                requestOptions
+            );
+            try {
+                logger.info(searchResponse);
+                // check match
+                ElasticsearchAssertions.assertSearchHits(searchResponse, id);
+            } finally {
+                searchResponse.decRef();
+            }
+
+            assertThat(
+                expectThrows(ResponseException.class, () -> client().performRequest(new Request("GET", "/" + index + "/_doc/" + id)))
+                    .getMessage(),
+                containsString("get operations not allowed on a legacy index")
+            );
+
+            // check that shards are skipped based on non-matching date
+            searchResponse = search(
+                index,
+                SearchSourceBuilder.searchSource().query(QueryBuilders.rangeQuery("create_date").from("2020-02-01")),
+                requestOptions
+            );
+            try {
+                logger.info(searchResponse);
+                assertEquals(0, searchResponse.getHits().getTotalHits().value());
+                assertEquals(numberOfShards, searchResponse.getSuccessfulShards());
+                // TODO the following is a failure tracked in https://github.com/elastic/elasticsearch/issues/115631
```

Comment:
This requires adaption once  https://github.com/elastic/elasticsearch/issues/115631 or the corresponding PR ([Fix OldRepositoryAccessIT testOldRepoAccess #117649](https://github.com/elastic/elasticsearch/pull/117649)) is closed.